### ISSUE TITLE
Testnet128 failure

### ISSUE THREAD

**Issue by lrettig** [spacemeshos/go-spacemesh#2409]
Tortoise began failing on tn128 around 11:00 UTC time yesterday, 5/2. The proximate cause of failure was the arrival of late blocks created by a single node with the node_id `5ea67`. Around 27 of these arrived on May 1, then another ~225 arrived beginning around 10:30 UTC on 5/2. I don't know the exact sequence of events that this triggered, but here's what I see in the logs:

layer 6954: the number of blocks being produced drops, from 40-50 per layer to < 30
layer 6956: this is the last layer tortoise was able to verify
layer 6958: even fewer blocks are produced
layer 6960: no blocks are produced
the following three layers: very few blocks are produced
layers 6956-6970: tortoise is unable to find a "minimal good layer"
layer 6971: tortoise starts finding a "minimal good layer" again, and block production recovers, but tortoise remains unable to verify any layers

AFAICT, (old) Tortoise is stuck because there aren't enough votes to approve any layer pattern. More recent patterns appear to be trying to vote against blocks that were contextually valid as of the last few validated layers, but the vote count isn't sufficient to cross the threshold. Here's what this looks like in the logs:

```
2021-05-02T09:28:12.814-0400    WARN    c992b.trtl              voting against block not in pattern view        {"node_id": "c992b21649d2a34385821de2bc179bcb940b37d5d385c514d865d385ed983ae3", "votingpattern": "patternID 3288270487 for layer 6984", "block_id_and_layer": "block_id 010e8fc091 layer 6956", "block_id": "010e8fc091", "layer_id": 6956, "pbase": "patternID 4094876152 for layer 6956", "vote_vector": "[0 371]"}
2021-05-02T09:28:12.814-0400    DEBUG   c992b.trtl              block voting based on global opinion of pattern {"node_id": "c992b21649d2a34385821de2bc179bcb940b37d5d385c514d865d385ed983ae3", "votingpattern": "patternID 3288270487 for layer 6984", "block_id_and_layer": "block_id 010e8fc091 layer 6956"}
2021-05-02T09:28:12.814-0400    DEBUG   c992b.trtl              global opinion  {"node_id": "c992b21649d2a34385821de2bc179bcb940b37d5d385c514d865d385ed983ae3", "vote_vector": "[0 371]", "threshold": "700"}
2021-05-02T09:28:12.814-0400    DEBUG   c992b.trtl              block support below threshold, block abstains from vote on pattern, pattern is incomplete and will not be promoted      {"node_id": "c992b21649d2a34385821de2bc179bcb940b37d5d385c514d865d385ed983ae3", "votingpattern": "patternID 3288270487 for layer 6984", "block_id": "010e8fc091", "layer_id": 6956, "vote_vector": "[0 0]"}
```

I found at least one case of a miner voting against a block it created, probably because the block was late. (block 4dd1da26ff created by miner-71 (96f69))

I strongly suspect #2405 is part of the story: the influx of late blocks effectively DoSed miners, which got stuck trying to process them and could not send outbound blocks on time or receive new ones.

Note that the log in https://github.com/spacemeshos/bug-reports/issues/5 is from the same miner, but this log ends before the chaos began, and I don't see any indication that any blocks at all were created by the miner in this log.

**Comment by lrettig** [comment]
I examined a later set of logs from the same node. There's a 24 hr gap between the two sets of logs, including the period of the "attack", so I still don't know exactly what happened, but in the later log I see that the node did begin to produce blocks. It appears to have been on its own fork, or in any case thought that it was the only active node. It saw an active set of size 1, and accordingly it began to produce ~50 blocks per layer. This seems consistent with what we saw on the managed miners: a sudden flood of blocks from one node.

Accordingly, here is my new diagnosis, in a nutshell: a node had severe p2p connection problems (for reasons we still don't understand). It was connected to peers, but it didn't see any blocks or ATXs and thought it was the only miner. It sent its own ATX and one epoch later it began to produce lots of blocks. Even though the node saw no inbound blocks, its blocks _did_ make it outbound and propagated across the network. This likely happened in a small number of short bursts - likely after the node was rebooted and briefly had live peers. This explosion of new blocks, in combination with the slow block initialization/processing times we're seeing, likely triggered an effective DoS attack against the rest of the network.

CC @noamnelke @antonlerner
